crates/subspace-solving/src/lib.rs

```diff
@@ -43,10 +43,12 @@ mod construct_uint {
     use subspace_core_primitives::PieceIndexHash;
 
     uint::construct_uint! {
-        /// Distance to piece index hash from farmer identity
-        pub struct PieceDistance(4);
+        pub struct U256(4);
     }
 
+    /// Distance to piece index hash from farmer identity
+    pub type PieceDistance = U256;
+
     impl PieceDistance {
```

Comment:
Shouldn't these impls be on `U256` now? They are effectively on `U256` even though it might seem that it isn't the case.

Reply:
Yeah, but semantically they are tied, so I decided to go with that.

Reply:
Let's not make it confusing and do impls on original types. `PieceDistance` is farmer-specific type alias and should likely be moved there altogether. I think this crate (and its users) should only need `U256`.

Reply:
`sp-consensus-subspace` should just use `U256`